Index out of bounds

```
Caused by: java.lang.RuntimeException: Failed to emit type goog.ui.ac.Renderer
	at com.google.javascript.cl2dts.DeclarationGenerator$TreeWalker.visitType(DeclarationGenerator.java:725)
	at com.google.javascript.cl2dts.DeclarationGenerator$TreeWalker.visitUnionType(DeclarationGenerator.java:773)
	at com.google.javascript.cl2dts.DeclarationGenerator$TreeWalker.access$1100(DeclarationGenerator.java:388)
	at com.google.javascript.cl2dts.DeclarationGenerator$TreeWalker$1.caseUnionType(DeclarationGenerator.java:606)
	at com.google.javascript.cl2dts.DeclarationGenerator$TreeWalker$1.caseUnionType(DeclarationGenerator.java:566)
	at com.google.javascript.rhino.jstype.UnionType.visit(UnionType.java:601)
	at com.google.javascript.cl2dts.DeclarationGenerator$TreeWalker.visitType(DeclarationGenerator.java:723)
	... 24 more
Caused by: java.lang.StringIndexOutOfBoundsException: String index out of range: -1
	at java.lang.String.substring(String.java:1918)
	at com.google.javascript.cl2dts.DeclarationGenerator$TreeWalker.getRelativeName(DeclarationGenerator.java:401)
	at com.google.javascript.cl2dts.DeclarationGenerator$TreeWalker.access$1000(DeclarationGenerator.java:388)
	at com.google.javascript.cl2dts.DeclarationGenerator$TreeWalker$1.caseObjectType(DeclarationGenerator.java:597)
	at com.google.javascript.cl2dts.DeclarationGenerator$TreeWalker$1.caseObjectType(DeclarationGenerator.java:566)
	at com.google.javascript.rhino.jstype.ObjectType.visit(ObjectType.java:524)
	at com.google.javascript.cl2dts.DeclarationGenerator$TreeWalker.visitType(DeclarationGenerator.java:723)

```

--- a/src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java
+++ b/src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java
@@ -398,6 +398,7 @@ public class DeclarationGenerator {
 
     private String getRelativeName(ObjectType objectType) {
       String name = objectType.getDisplayName();
+      if (name.equals(namespace)) return name;
       return !name.startsWith(namespace) ? name : name.substring(namespace.length() + 1);
     }
 

--- a/src/test/java/com/google/javascript/cl2dts/types.d.ts
+++ b/src/test/java/com/google/javascript/cl2dts/types.d.ts
@@ -4,7 +4,6 @@ declare namespace ಠ_ಠ.cl2dts_internal.types {
   var c : string ;
   var d : Object ;
   var e : any [] ;
-  var arrayMissingTypeParam : any [] ;
   var functionAndUnion : (a : number , b : any ) => any ;
   var recordType : { a : string , b : any } ;
   var j : { [ n: number ]: string } ;

--- a/src/test/java/com/google/javascript/cl2dts/types.js
+++ b/src/test/java/com/google/javascript/cl2dts/types.js
@@ -15,9 +15,6 @@ types.d = {};
 /** @type {Array<?>} */
 types.e = [];
 
-/** @type {Array} */
-types.arrayMissingTypeParam = [];
-
 /** @type {null|function(number, ?):?} handler */
 types.functionAndUnion = null;
 

